src/frontend/src/handler/util.rs

```diff
@@ -441,4 +444,55 @@ mod tests {
             "1969-12-31 23:59:59.999999+00:00"
         );
     }
+
+    fn to_object_name(s: &str) -> ObjectName {
+        ObjectName(vec![Ident::new_unchecked(s)])
+    }
+
+    #[test]
+    fn test_redact() {
+        use risingwave_sqlparser::ast::utils::SqlOptionVecSerializer;
+        use serde::Serialize;
+
+        let p = S3PropertiesCommon {
+            region_name: "region".to_string(),
+            bucket_name: "bucket".to_string(),
+            match_pattern: Some("pattern".into()),
+            access: None,
+            secret: Some(SecretString::new("123")),
+            endpoint_url: None,
+        };
+        let mut s = SqlOptionVecSerializer::default();
+        p.serialize(&mut s).unwrap();
+        let sql_options: Vec<SqlOption> = s.into();
+        assert_eq!(
+            sql_options,
+            vec![
+                SqlOption {
+                    name: to_object_name("s3.region_name"),
+                    value: Value::SingleQuotedString("region".into())
+                },
+                SqlOption {
+                    name: to_object_name("s3.bucket_name"),
+                    value: Value::SingleQuotedString("bucket".into())
+                },
+                SqlOption {
+                    name: to_object_name("match_pattern"),
+                    value: Value::SingleQuotedString("pattern".into())
+                },
+                SqlOption {
+                    name: to_object_name("s3.credentials.access"),
+                    value: Value::Null
+                },
+                SqlOption {
+                    name: to_object_name("s3.credentials.secret"),
+                    value: Value::SingleQuotedString("[REDACTED alloc::string::String]".into())
```

Comment:
The redacted display does not look that good as `alloc::string::String` seems verbose. 🤣 It's not a big deal as we can replace the internal implementation of `SecretString` in the future.